Update encryption ARM template doc

simonesavi opened the pull request:
List of updates Linux:
	- Rewrite Extension Schemata session
	- On the ARM template v1.1 example:
		○ Removed parameter 'SequenceVersion'. Looking the last extension version (1.1.0.47) it seems to be no longer present
		○ Updated apiVersion to last available
		○ Inserted 'autoUpgradeMinorVersion'
		○ Re-ordered to better reading
	- On the ARM template v0.1 example:
		○ Updated apiVersion to last available
		○ Re-ordered to better reading
	- Property values
		○ Inserted missing values
		○ Insert references to v0.1 and optional sessions
	- Inserted reference to ARM template example based on v1.1
	- Removed 'Azure CLI deployment' and add link Next step that has a full description of all encryption methods
	- Fixed link in Troubleshoot

List of updates Windows:
- Prerequisites: Reviewed links and removed references to linux environment
- Extension schemata: Aligned to Linux proposition
- Schema: Various updates
- Property values: Review tables and values. Removed references to linux
- Template deployment: New links
- Removed Azure CLI deployment

Updates on azure-disk-encryption-vms-vmss.md:
- Add links to ARM template (virtual machine extension)

simonesavi committed:
Review entire content

List of updates:
	- Rewrite Extension Schemata session
	- On the ARM template v1.1 example:
		○ Removed parameter 'SequenceVersion'. Looking the last extension version (1.1.0.47) it seems to be no longer present
		○ Updated apiVersion to last available
		○ Inserted 'autoUpgradeMinorVersion'
		○ Re-ordered to better reading
	- On the ARM template v0.1 example:
		○ Updated apiVersion to last available
		○ Re-ordered to better reading
	- Property values
		○ Inserted missing values
		○ Insert references to v0.1 and optional sessions
	- Inserted reference to ARM template example based on v1.1
	- Removed 'Azure CLI deployment' and add link Next step that has a full description of all encryption methods
	- Fixed link in Troubleshoot

simonesavi committed:
Update azure-disk-encryption-vms-vmss.md

Add links to ARM template (virtual machine extension)

simonesavi committed:
Update azure-disk-encryption-vms-vmss.md

Remove create word from ARM template link. The linked ARM template is only to enable encryption

simonesavi committed:
Update azure-disk-enc-linux.md

Fixed some typos
re-insert SequenceVersion

simonesavi committed:
Update azure-disk-enc-windows.md

List of changes:
- Prerequisites: Reviewed links and removed references to linux environment
- Extension schemata: Aligned to Linux proposition
- Schema: Various updates
- Property values: Review tables and values. Removed references to linux
- Template deployment: New links
- Removed Azure CLI deployment

simonesavi committed:
Update azure-disk-enc-windows.md

Typo

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-windows.md:
```diff
@@ -24,15 +24,19 @@ Azure Disk Encryption leverages BitLocker to provide full disk encryption on Azu
 
-There are two schemata for the Windows AzureDiskEncryption extension: v2.2, a newer, recommended schema that does not use Azure Active Directory (AAD) properties, and v1.1, an older schema that requires AAD properties. You must use the schema version corresponding to the extension you are using: schema v2.2 for the AzureDiskEncryption extension version 2.2, schema v1.1 for the AzureDiskEncryption ...
+There are two versions of extension schema for Azure Disk Encryption (ADE):
+- v2.2 - A newer recommended schema that does not use Azure Active Directory (AAD) properties.
```
There is a nuance here on extension version:  

Linux - AzureDiskEncryptionForLinux - is 0.1 (AAD) /1.1 (No-AAD)
Windows - AzureDiskEncryption is 1.1 (AAD) /2.2  (No-AAD)

simonesavi replied:
Hi @ejarvi,
I have just re-check content of both file (linux and windows) and versions seams to be aligned to your suggestion.

ejarvi replied:
This page is dedicated specifically to Windows now - so yes the alignment is correct.  👍

ejarvi commented on articles/security/fundamentals/azure-disk-encryption-vms-vmss.md:
```diff
@@ -25,6 +25,7 @@ The following articles provide guidance for encrypting Linux virtual machines.
 - [Create and encrypt a Linux VM with Azure CLI](../../virtual-machines/linux/disk-encryption-cli-quickstart.md)
 - [Create and encrypt a Linux VM with Azure PowerShell](../../virtual-machines/linux/disk-encryption-powershell-quickstart.md)
 - [Create and encrypt a Linux VM with the Azure portal](../../virtual-machines/linux/disk-encryption-portal-quickstart.md)
+- [Encrypt a Linux VM with the ARM template](../../virtual-machines/extensions/azure-disk-enc-linux.md)
```
Consider changing title to "Azure Disk Encryption Extension Schemata for Linux" ? 
"the ARM template" makes it sounds like there is one single ARM template but ARM templates are customizable per users.   Also the link that it goes to describes the extension schema that is used by REST API, ARM templates, even the powershell/cli commands wrap this when they make calls to add extensions - the schema is not specific to ARM templates.  
There are some examples of ARM templates on Github here, although it is outside of docs.microsoft.com :   https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm

ejarvi replied:
@simonesavi could you update the PR with this change?

simonesavi replied:
@ejarvi done

ejarvi commented on articles/security/fundamentals/azure-disk-encryption-vms-vmss.md:
```diff
@@ -47,6 +48,7 @@ The following articles provide guidance for encrypting Windows virtual machines.
 - [Create and encrypt a Windows VM with Azure CLI](../../virtual-machines/windows/disk-encryption-cli-quickstart.md)
 - [Create and encrypt a Windows VM with Azure PowerShell](../../virtual-machines/windows/disk-encryption-powershell-quickstart.md)
 - [Create and encrypt a Windows VM with the Azure portal](../../virtual-machines/windows/disk-encryption-portal-quickstart.md)
+- [Encrypt a Windows VM with the ARM template](../../virtual-machines/extensions/azure-disk-enc-windows.md)
```
Consider changing title to "Azure Disk Encryption Extension Schemata for Windows"  (the link it goes to describes the schema for Windows, it doesn't describe an ARM template)

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-linux.md:
```diff
@@ -130,40 +139,51 @@ Using `AADClientCertificate`:
 ## Template deployment
 
-For an example of template deployment, see [Enable Encryption on a running Linux VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm).
+For an example of template deployment based on schema v1.1, see [Enables Encryption on a running Linux VM with no AD required](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm-without-aad).
```
Suggest changing wording of the link from:
`see [Enables Encryption on a running Linux VM with no AD required](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm-without-aad).
`
to: 

`see the Azure Quickstart Template [201-encrypt-running-linux-vm-without-aad](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm-without-aad)`

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-linux.md:
```diff
@@ -130,40 +139,51 @@ Using `AADClientCertificate`:
-For an example of template deployment, see [Enable Encryption on a running Linux VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm).
+For an example of template deployment based on schema v1.1, see [Enables Encryption on a running Linux VM with no AD required](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm-without-aad).
+
+For an example of template deployment based on schema v0.1, see [Enable Encryption on a running Linux VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm).
```
Suggest changing:

`see [Enable Encryption on a running Linux VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm).`

to:

`see Azure Quickstart Template [201-encrypt-running-linux-vm](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-linux-vm).`

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-linux.md:
```diff
@@ -130,40 +139,51 @@ Using `AADClientCertificate`:
-## Azure CLI deployment
+>[!WARNING]
+> - If you have previously used Azure Disk Encryption with Azure AD to encrypt a VM, you must continue use this option to encrypt your VM.
+> - When encrypting Linux OS volumes, the VM should be considered unavailable. We strongly recommend to avoid SSH logins while the encryption is in progress to avoid issues blocking any open files that will need to be accessed during the encryption process. To check progress, use the the [Get-AzVMDiskEncryptionStatus](/powershell/module/az.compute/get-azvmdiskencryptionstatus) PowerShell cmdlet o ...
```
Change: 

`use the the [Get-AzVMDiskEncryptionStatus]`

to: 

`use the [Get-AzVMDiskEncryptionStatus]`

Also - if using Powershell to check encryption status, using the most recent version of Azure Powershell is required.

simonesavi replied:
@ejarvi  done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-linux.md:
- Go to [Azure support]
- capitalize "S" in support:  "Microsoft Azure Support FAQ"

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-windows.md:
```diff
@@ -132,39 +137,44 @@ Using `AADClientCertificate`:
-Create a new encrypted Windows VM from gallery image](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-create-new-vm-gallery-image).
 
-## Azure CLI deployment
+For an example of template deployment based on schema v2.2, see [Enable encryption on a running Windows VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-windows-vm-without-aad).
```
`see Azure QuickStart Template [201-encrypt-running-windows-vm-without-aad](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-windows-vm-without-aad).`

ejarvi replied:
@simonesavi could we change the wording of this from " see Enable encryption on a running windows vm" to "see Azure QuickStart Template 201-encrypt..."

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-windows.md:
```diff
@@ -132,39 +137,44 @@ Using `AADClientCertificate`:
-## Azure CLI deployment
+For an example of template deployment based on schema v2.2, see [Enable encryption on a running Windows VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-windows-vm-without-aad).
+
+For an example of template deployment based on schema v1.1, see [Enable encryption on a running Windows VM](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-windows-vm).
```
`see Azure Quickstart Template [201-encrypt-running-windows-vm](https://github.com/Azure/azure-quickstart-templates/tree/master/201-encrypt-running-windows-vm).`

ejarvi replied:
@simonesavi same - can you update the PR to specify the exact template by name?

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-windows.md:
```diff
@@ -132,39 +137,44 @@ Using `AADClientCertificate`:
 
-Instructions can be found in the latest [Azure CLI documentation](/cli/azure/vm/encryption?view=azure-cli-latest). 
+>[!NOTE]
+> Also if `VolumeType` parameter is set to Data or All, data disks will be encrypted only if they are properly formatted. 
```
change: 
"is set to Data or All"

to:
"is set to All"  

(don't need to include this in the note, but on Windows it is recommended to use OS or All)

simonesavi replied:
@ejarvi done

ejarvi commented on articles/virtual-machines/extensions/azure-disk-enc-windows.md:
capital S  - "Microsoft Azure Support FAQ"

simonesavi replied:
@ejarvi done

ejarvi reviewed (commented):
Looking good.  I added some additional review comments, and I'm handing off to @msmbaldwin for final approval.

simonesavi committed:
Update azure-disk-encryption-vms-vmss.md

Changes:
Encrypt a Linux VM with the ARM template -> Azure Disk Encryption Extension Schemata for Linux
Encrypt a Windows VM with the ARM template -> Azure Disk Encryption Extension Schemata for Windows

simonesavi committed:
Update azure-disk-enc-linux.md

Changes:
- see [Enables Encryption on a running Linux VM with no AD required] -> see the Azure Quickstart Template [201-encrypt-running-linux-vm-without-aad]
- see [Enable Encryption on a running Linux VM] -> see Azure Quickstart Template [201-encrypt-running-linux-vm]
- typos

simonesavi committed:
Update azure-disk-enc-linux.md

Changes:
- Go to [Azure support]
- capitalize "S" in support: "Microsoft Azure Support FAQ"

simonesavi committed:
Update azure-disk-enc-windows.md

Changes:
- see [Enable encryption on a running Windows VM] -> see Azure QuickStart Template [201-encrypt-running-windows-vm-without-aad]
- see [Enable encryption on a running Windows VM] -> see Azure Quickstart Template [201-encrypt-running-windows-vm]
- Go to [Azure support]
- capitalize "S" in support: "Microsoft Azure Support FAQ"
-"is set to Data or All" -> "is set to All"
